types/bn.js/bn.js-tests.ts

```diff
@@ -1,23 +1,33 @@
 // test that it works as a require
-import BN = require('bn.js');
+import BN_require = require('bn.js');
+// test that it works as module import
+import * as BN_esm from 'bn.js';
+import { Endianness, IPrimeName } from 'bn.js';
 
-let bn = new BN(42);
-bn = bn.add(bn);
-bn.isZero();
-bn.byteLength;
+function runTests(BN: typeof BN_esm) {
+    let bn = new BN(42);
+    bn = bn.add(bn);
+    bn.isZero();
+    bn.byteLength;
 
-bn.toArrayLike(Buffer, 'le', 2);
-const test = new BN(1, 'le');
+    const endian: Endianness = 'le';
+    bn.toArrayLike(Buffer, endian, 2);
+    const test = new BN(1, endian);
 
-const ctx = BN.red('p224');
-ctx.prime.name;
+    const primeName: IPrimeName = 'p224';
+    const ctx = BN.red(primeName);
+    ctx.prime.name;
 
-const red = bn.toRed(ctx);
-const newRed = red.redAdd(new BN(1));
-newRed.cmp(bn);
-newRed.fromRed();
+    const red = bn.toRed(ctx);
+    const newRed = red.redAdd(new BN(1).toRed(ctx));
```

Comment:
Reviewer note:
This throws an error since at least v4.11.0 (the version indicated in the type def).
See: https://github.com/indutny/bn.js/blob/e670ab2360a6f37994014d85a07849c81817c1b0/lib/bn.js#L2849-L2852

Added the `.toRed` to fix the error.